planner/core/testdata/runtime_filter_generator_suite_out.json

```diff
@@ -0,0 +1,178 @@
+          "  └─Projection_8 1.00 mpp[tiflash]  test.t1.k1, test.t2.k1, test.t2.k2, test.t2.k3",
+          "    └─HashJoin_18 1.00 mpp[tiflash]  inner join, equal:[eq(Column#7, test.t2.k1)], runtime filter:0[IN] <- Column#7",
+          "      ├─ExchangeReceiver_14(Build) 1.00 mpp[tiflash]  ",
+          "      │ └─ExchangeSender_13 1.00 mpp[tiflash]  ExchangeType: Broadcast, Compression: FAST",
+          "      │   └─Projection_11 1.00 mpp[tiflash]  test.t1.k1, plus(test.t1.k1, 1)->Column#7",
+          "      │     └─TableFullScan_12 1.00 mpp[tiflash] table:t1 keep order:false",
+          "      └─TableFullScan_15(Probe) 1.00 mpp[tiflash] table:t2 keep order:false, runtime filter:0[IN] -> test.t2.k1"
+        ]
+      },
+      {
+        "SQL": "select /*+ broadcast_join(t2, t1), hash_join_build(t2) */ * from t2, (select k1, k1+1 as k11 from t1) t1 where t1.k1=t2.k1; -- Only support origin column k1",
+        "Plan": [
+          "TableReader_21 1.00 root  MppVersion: 1, data:ExchangeSender_20",
+          "└─ExchangeSender_20 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─HashJoin_19 1.00 mpp[tiflash]  inner join, equal:[eq(test.t2.k1, test.t1.k1)], runtime filter:0[IN] <- test.t2.k1",
+          "    ├─ExchangeReceiver_13(Build) 1.00 mpp[tiflash]  ",
+          "    │ └─ExchangeSender_12 1.00 mpp[tiflash]  ExchangeType: Broadcast, Compression: FAST",
+          "    │   └─Selection_11 1.00 mpp[tiflash]  not(isnull(test.t2.k1))",
+          "    │     └─TableFullScan_10 1.00 mpp[tiflash] table:t2 pushed down filter:empty, keep order:false",
+          "    └─Projection_14(Probe) 1.00 mpp[tiflash]  test.t1.k1, plus(test.t1.k1, 1)->Column#7",
+          "      └─Selection_16 1.00 mpp[tiflash]  not(isnull(test.t1.k1))",
+          "        └─TableFullScan_15 1.00 mpp[tiflash] table:t1 pushed down filter:empty, keep order:false, runtime filter:0[IN] -> test.t1.k1"
+        ]
+      },
+      {
+        "SQL": "select /*+ hash_join_build(t2) */ * from t2, (select k1, k1+1 as k11 from t1) t1 where t1.k11=t2.k1; -- Doesn't support transform column k11",
+        "Plan": [
+          "TableReader_36 0.80 root  MppVersion: 1, data:ExchangeSender_35",
+          "└─ExchangeSender_35 0.80 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─HashJoin_29 0.80 mpp[tiflash]  inner join, equal:[eq(test.t2.k1, Column#7)]",
+          "    ├─ExchangeReceiver_33(Build) 1.00 mpp[tiflash]  ",
+          "    │ └─ExchangeSender_32 1.00 mpp[tiflash]  ExchangeType: Broadcast, Compression: FAST",
+          "    │   └─Selection_31 1.00 mpp[tiflash]  not(isnull(test.t2.k1))",
+          "    │     └─TableFullScan_30 1.00 mpp[tiflash] table:t2 pushed down filter:empty, keep order:false",
+          "    └─Projection_34(Probe) 0.80 mpp[tiflash]  test.t1.k1, plus(test.t1.k1, 1)->Column#7",
+          "      └─Selection_26 0.80 mpp[tiflash]  not(isnull(plus(test.t1.k1, 1)))",
```

Comment:
Seems we have a todo here.
the simple expression can be deconstructed to `k1 =  values-1`
or
just let Tiflash support expression choose when doing lazy materialization.

Reply:
Yes, we need to combine these two optimizations later. This is a todo